lms/static/js/spec/student_account/account_settings_fields_spec.js

```diff
@@ -0,0 +1,320 @@
+                    var data = {'name': fields[i][2]};
+
+                    var view = new fields[i][0](fieldData).render();
+
+                    // Initially the help message is shown
+                    expectMessageContains(view, fieldData.helpMessage);
+
+                    view.$(selector).val(data.name).change();
+                    // When the value in the field is changed
+                    expect(view.fieldValue()).toBe(fields[i][2]);
+                    expectMessageContains(view, view.inProgressMessage);
+                    expectAjaxRequestWithData(data);
+
+                    AjaxHelpers.respondWithNoContent(requests);
+                    // When server returns success.
+                    expectMessageContains(view, view.successMessagePrefix);
+
+                    view.$(selector).val(data.name).change();
+                    AjaxHelpers.respondWithError(requests, 500);
+                    // When server returns a 500 error
+                    expectMessageContains(view, view.errorMessage);
+
+                    view.$(selector).val('').change();
+                    AjaxHelpers.respondWithError(requests, 400, createErrorMessage(fieldData.valueAttribute, validationError));
+                    // When server returns a validation error
+                    expectMessageContains(view, view.invalidInputMessagePrefix);
+                    expectMessageContains(view, validationError);
+                }
+            });
+
+            it("correctly renders LinkFieldView", function() {
+                var fieldData = createFieldData(AccountSettingsFieldViews.LinkFieldView, {
+                    title: 'Title',
+                    linkTitle: 'Link title',
+                    helpMessage: 'Click the link.'
+                });
+                var view = new AccountSettingsFieldViews.LinkFieldView(fieldData).render();
+                expectTitleAndMessageToBe(view, fieldData.title, fieldData.helpMessage);
+                expect(view.$('.account-settings-field-value > a').text().trim()).toBe(fieldData.linkTitle);
+            });
+
+            it("sends request to reset password on clicking link in PasswordFieldView", function() {
+                requests = AjaxHelpers.requests(this);
+
+                var fieldData = createFieldData(AccountSettingsFieldViews.PasswordFieldView, {
+                    linkHref: '/password_reset'
+                });
+
+                var view = new AccountSettingsFieldViews.PasswordFieldView(fieldData).render();
+                view.$('.account-settings-field-value > a').click();
+                AjaxHelpers.expectRequest(requests, 'POST', '/password_reset', "email=legolas%40woodland.middlearth");
+                AjaxHelpers.respondWithJson(requests, {"success": "true"})
+                expectMessageContains(view,
+                    "Complete you password reset using the confirmation link emailed to legolas@woodland.middlearth."
```

Comment:
Text:

We've sent a message to your email address. Click the link in the message to reset your password.